code/modules/awaymissions/capture_the_flag.dm

```diff
@@ -290,3 +302,35 @@
 
 /area/ctf/flag_room2
 	name = "Flag Room B"
+
+
+//Control Point
+
+/obj/machinery/control_point
+	name = "control point"
+	desc = "You should capture this."
+	icon = 'icons/obj/machines/dominator.dmi'
+	icon_state = "dominator"
+	var/obj/machinery/capture_the_flag/controlling
+	var/team = "none"
+	var/point_rate = 1
+
+/obj/machinery/control_point/process()
+	if(controlling)
+		controlling.control_points += point_rate
+		if(controlling.control_points >= controlling.control_points_to_win)
+			controlling.victory()
+
+/obj/machinery/control_point/attackby(mob/user, params)
+	capture(user)
+
+/obj/machinery/control_point/attack_hand(mob/user)
+	capture(user)
+
+/obj/machinery/control_point/proc/capture(mob/user)
+	if(do_after(user, 30, target = src))
+		for(var/obj/machinery/capture_the_flag/CTF in machines)
+			if(CTF.ctf_enabled && user.faction == CTF.team)
```

Comment:
It works, but it violates a cardinal assumption of faction code, so it would runtime to death if you did anything else faction related (like ctf AI-controlled mobs, on-station ctf, or anything really). Plus it's not even significantly easier, considering you just gotta replace the `faction = blah` with `faction += blah` and `faction = null` with `faction -= blah`. Making faction a list for everything except ctf is just a recipe for esoteric runtimes and mystery bugs.